### PR TITLE
Story/sc 1339 update poetry

### PR DESCRIPTION
**Pull request by Fabian6781**
## Jira Story
[SC-1339](https://epcpower.atlassian.net/browse/SC-1339)

## About
Update Poetry version from 1.1.15 to 1.5.1.

## Associated Pull Requests
N/A

## Validation

[SC-1339]: https://epcpower.atlassian.net/browse/SC-1339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pyproject.toml
Comment (gordon-epc) [inline pyproject.toml:1]: `build-backend = "poetry.core.masonry.api"`
to
`build-backend = "poetry_dynamic_versioning.backend"`

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: pyproject.toml
Comment (Fabian6781) [reply pyproject.toml:1]: Done.

File: pyproject.toml [hunk 1]
```diff
@@ -8,19 +11,19 @@ license = "GPL-2.0-or-later"
 
 [tool.poetry-dynamic-versioning]
 #modifies epyqlib/__init__.py
```
[the remark this answers is not in the document]
Comment (gordon-epc) [reply pyproject.toml:13]: Alex is right to check, as this doesn't appear to be working. The file `epyqlib/__init__.py` should have the `__version__` line modified to look something like in the built artifact:
`
__version__ = "2022.10.post25.dev0+0166177"
`
but is instead left as 
`
__version__ = "0.0.0"
`

@Fabian6781 Look at the poetry dynamic versioning instructions. It likely has something to do with `tool.poetry-dynamic-versioning.substitution`, but it is not clear how that should be configured in the `pyproject.toml`.

[the remark this answers is not in the document]
Comment (Fabian6781) [reply pyproject.toml:13]: Resolved.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -56,21 +56,21 @@ jobs:
...
+          poetry config installer.max-workers 1    # https://github.com/python-poetry/poetry/issues/7611
 
           poetry install
-
           poetry --version
       - name: python commands
         run: |
           set -vx
           poetry run pip --version
+          poetry run poetry-dynamic-versioning
```
Comment (gordon-epc) [inline .github/workflows/ci.yml:67]: This is the old way that we are trying to remove.
```suggestion
```

File: pyproject.toml [hunk 3]
```diff
@@ -80,6 +85,7 @@ pytest-twisted = "<=1.13.4"
 pytest-xvfb = "1.2.0"
 twine = "1.13.0"
 pyqt5-tools = { version = "5.13.2.1.6rc1", markers = "sys_platform == 'win32'" }
+poetry-dynamic-versioning = "^0.13.1"
```
Comment (gordon-epc) [inline pyproject.toml:88]: This is the old way that we are trying to remove.
```suggestion
```

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Fabian6781) [reply .github/workflows/ci.yml]: Resolved.

File: pyproject.toml (hunk 3 above)
Comment (Fabian6781) [reply pyproject.toml]: Resolved.

File: pyproject.toml [hunk 4]
```diff
@@ -8,19 +11,22 @@ license = "GPL-2.0-or-later"
 
 [tool.poetry-dynamic-versioning]
 #modifies epyqlib/__init__.py
-enable = false
+enable = true
 vcs = "git"
 style = "pep440"
 
+#[tool.poetry-dynamic-versioning.substitution]
```
[the remark this answers is not in the document]
Comment (Fabian6781) [reply pyproject.toml]: Removed.

**Comment by Fabian6781** [comment]
@gordon-epc , This PR needs to be merged first, and then PM will point to epyqlib develop.

**Review by gordon-epc — APPROVED** [review]

**Pull request merged**